### PR TITLE
(BOLT-772) Allow failure when removing Bolt Apply temp directory

### PR DESCRIPTION
**Pull request by MikaelSmith**
Some providers keep handle open to files in their module after running.
On Windows this prevents deleting the temporary modulepath we create
when applying a catalog. Allow this to fail and return a message over
stderr (which will be logged at debug level).

### CODE REVIEW

**Comment by MikaelSmith** [comment]
When we start caching module code on targets, this will probably be a non-issue.

File: libexec/apply_catalog.rb [hunk 1]
```diff
@@ -42,39 +43,43 @@
...
+  catalog.environment_instance = env
+
+  configurer = Puppet::Configurer.new
+  configurer.run(catalog: catalog, report: report, pluginsync: false)
+end
+
+puts JSON.pretty_generate(report.to_data_hash)
+exit_code = report.exit_status != 1
+
+begin
```
Comment (adreyer) [inline libexec/apply_catalog.rb]: Should this be in a finally?

Comment (MikaelSmith) [reply libexec/apply_catalog.rb]: That would make sense...

File: libexec/apply_catalog.rb [hunk 2]
```diff
@@ -43,7 +44,8 @@
 Puppet[:default_file_terminus] = :file_server
 
 exit_code = 0
-Dir.mktmpdir do |moduledir|
+begin
+  moduledir = Dir.mktmpdir
```
Comment (adreyer) [inline libexec/apply_catalog.rb]: Should this be outside the begin so we get the 'proper' error when Dirmktmpdir fails?

Comment (MikaelSmith) [reply libexec/apply_catalog.rb]: Well, we shouldn't try to remove it if we don't create it in the first place.

**Pull request merged**